flashlight/nn/modules/BatchNorm.cpp

```diff
@@ -59,27 +59,32 @@ Variable BatchNorm::forward(const Variable& input) {
 
   return batchnorm(
       input,
-      params_.empty() ? Variable() : params_[0],
-      params_.empty() ? Variable() : params_[1],
-      runningMean_,
-      runningVar_,
+      params_[0],
+      params_[1],
+      params_[2],
+      params_[3],
       featAxes_,
       train_ || (!trackStats_),
       avg_factor,
       epsilon_);
 }
 
 void BatchNorm::initialize() {
+  Variable wt, bs;
+  if (affine_) {
+    wt = uniform(featSize_, 0.0, 1.0, af::dtype::f32, true);
+    bs = constant(0.0, featSize_, af::dtype::f32, true);
+  }
+  else {
+    wt = Variable();
+    bs = Variable();
+  }
+
   if (trackStats_) {
     runningMean_ = constant(0.0, featSize_, af::dtype::f32, false);
     runningVar_ = constant(1.0, featSize_, af::dtype::f32, false);
   }
-
-  if (affine_) {
-    auto wt = uniform(featSize_, 0.0, 1.0, af::dtype::f32, true);
-    auto bs = constant(0.0, featSize_, af::dtype::f32, true);
-    params_ = {wt, bs};
-  }
+  params_ = {wt, bs, runningMean_, runningVar_};
```

Comment:
I have looked at PyTorch, TF implementations. This doesn't seem to be the standard. 

Can we just have functions called `setRunningMean`, `getRunningMean`, `setRunningVar`, `getRunningVar`. I guess that would work for you ?

Reply:
Yes, it will be good. I will revise this to add functions you said soon.

Reply:
@minty99 any update here? Would be nice to rebase and land these.

Reply:
@jacobkahn I'm sorry but I was busy. I will work on this again soon.